### ISSUE TITLE
Misleading suggestion for unused shorthand field pattern

### ISSUE THREAD

**Issue by andersk** [rust-lang/rust#46866]
```rust
use std::ops::Range;
fn range_end(Range { start, end }: Range<u32>) -> u32 {
    end
}
```

gives

```
warning: unused variable: `start`
 --> test.rs:2:22
  |
2 | fn range_end(Range { start, end }: Range<u32>) -> u32 {
  |                      ^^^^^
  |
  = note: #[warn(unused_variables)] on by default
  = note: to avoid this warning, consider using `_start` instead
```

You can’t just change `start` to `_start` here, of course—we should suggest `start: _` instead.

**Comment by Mark-Simulacrum** [comment]
#47390 has more discussion, so I'll close in favor of that. Thanks!
